### PR TITLE
Initial setup for the project

### PR DESCRIPTION
**Pull request by Jonesus**
This is an initial setup proposal for the mobile application

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module.exports = {
+  root: true,
+  extends: [
+    '@react-native-community',
+    'airbnb-typescript',
+    'prettier',
+    'prettier/@typescript-eslint',
+    'prettier/react',
+  ],
+  plugins: ['react-hooks', '@typescript-eslint/eslint-plugin'],
```
Comment (pheis) [inline .eslintrc.js]:
```suggestion
  plugins: ['react-hooks', '@typescript-eslint/eslint-plugin',
    "import",
    "react",
    "react-hooks",
    "react-native"
],
```

Comment (pheis) [inline .eslintrc.js]: import plugin is super nice with images!

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    'import/resolver': {
+      node: {
+        paths: ['./'],
+        extensions: ['.js', '.jsx', '.ts', '.tsx', '.d.ts'],
+      },
+    },
+  },
+  rules: {
+    '@typescript-eslint/indent': ['error', 2],
+    '@typescript-eslint/no-use-before-define': ['off'],
```
Comment (pheis) [inline .eslintrc.js]:
```suggestion
    '@typescript-eslint/no-use-before-define': ['off'],
    '@typescript-eslint/camelcase': 'off',
    '@typescript-eslint/explicit-function-return-type': 'off',
    '@typescript-eslint/no-unused-vars': [
      'error',
      {
        'args': 'after-used',
        'argsIgnorePattern': '^_$',
        'ignoreRestSiblings': true
      }
    ],
    '@typescript-eslint/restrict-plus-operands': 'error',
```

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    'react-hooks/rules-of-hooks': 'error',
+    'react-hooks/exhaustive-deps': ['off'],
+    'jsx-quotes': ['error', 'prefer-double'],
+    quotes: [2, 'single'],
+    'max-len': [
+      'error',
+      {
+        code: 80,
+        tabWidth: 2,
+      },
```
Comment (pheis) [inline .eslintrc.js]: prettier does this already

File: package.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    "eslint-config-airbnb-typescript": "6.0.0",
+    "eslint-config-prettier": "6.5.0",
+    "eslint-plugin-import": "2.18.2",
+    "eslint-plugin-jsx-a11y": "6.2.3",
+    "eslint-plugin-react": "7.16.0",
+    "eslint-plugin-react-hooks": "2.2.0",
+    "jest": "24.9.0",
+    "metro-react-native-babel-preset": "0.56.0",
+    "prettier": "1.18.2",
+    "react-test-renderer": "16.9.0",
```
Comment (pheis) [inline package.json]: "react-native-svg-asset-plugin": "0.3",
I can do another PR for this one : P

**Comment by pheis** [comment]
This looks good! We can iterate more as we move on : )

File: .eslintrc.js (hunk 3 above)
Comment (Jonesus) [reply .eslintrc.js]: Yeah but then the editor will complain so let's keep them

File: package.json (hunk 4 above)
Comment (Jonesus) [reply package.json]: Please : ) Or maybe only after we really need those

**Pull request merged**